lib/generators/mi/mi_generator.rb

```diff
@@ -1,113 +1,65 @@
+          puts Mi::VERSION
+          return
+        end
 
-  source_root File.expand_path('../templates', __FILE__)
+        migration_template('migration.rb.erb', "db/migrate/#{destination}.rb")
+      end
 
-  def doing
-    if arguments.include?('--version')
-      puts Mi::VERSION
-      return
-    end
 
-    migration_template('migration.rb.erb', "db/migrate/#{destination}.rb")
-  end
+      private
 
 
-  private
+      # @param [String] col +COL_NAME:TYPE:{OPTIONS}
+      def to_method(table, col)
+        info = parse_column(col)
+        res = "#{Methods[info[:method]]} :#{table}, :#{info[:name]}"
 
-  def arguments
-    @_initializer[0..1].flatten
-  end
+        return res unless info[:type]
+        res << ", :#{info[:type]}"
 
-  def arg_groups
-    @arg_groups ||= (
-      args = arguments.reject{|x| x.start_with?('--')}
+        return res unless info[:options]
+        res << ", #{info[:options][1..-2].gsub(':', ': ').gsub(',', ', ')}"
 
-      current = nil
-      res = args.group_by do |a|
-        if %w[+ - %].include? a[0]
-          current
-        else
-          current = a
-          nil
-        end
+        res
       end
-      res.delete(nil)
-      res
-    )
-  end
 
-  # TODO: parse options
-  # @param [String] col +COL_NAME:TYPE:{OPTIONS}
-  def parse_column(col)
-    sc = StringScanner.new(col)
-    name = sc.scan(/[^:]+/)
-    sc.scan(/:/)
-    type = sc.scan(/[^:]+/)
-    sc.scan(/:/)
-    options = sc.scan(/\{.+\}$/)
-
-    method = name[0]
-    name = name[1..-1]
-    {name: name, type: type, options: options, method: method}
-  end
-
-  # @param [String] col +COL_NAME:TYPE:{OPTIONS}
-  def to_method(table, col)
-    info = parse_column(col)
-    res = "#{Methods[info[:method]]} :#{table}, :#{info[:name]}"
-
-    return res unless info[:type]
-    res << ", :#{info[:type]}"
-
-    return res unless info[:options]
-    res << ", #{info[:options][1..-2].gsub(':', ': ').gsub(',', ', ')}"
-
-    res
-  end
-
-  def to_dest(col)
-    parsed = parse_column(col)
-    verb =
-      case parsed[:method]
-      when '+'
-        'add'
-      when '-'
-        'remove'
-      when '%'
-        'change'
+      def to_dest(col)
+        parsed = parse_column(col)
+        verb =
+          case parsed[:method]
+          when '+'
+            'add'
+          when '-'
+            'remove'
+          when '%'
+            'change'
+          end
+        [verb, parsed[:name]]
       end
-    [verb, parsed[:name]]
-  end
 
-  def destination
-    table, columns = *arg_groups.first
-    c = {
-      '+' => 'to',
-      '-' => 'from',
-      '%' => 'of'
-    }[parse_column(columns.last)[:method]]
-    [
-      columns.map{|c| to_dest(c)}.inject{|sum, x| sum.concat(['and', x].flatten)},
-      c,
-      table.tableize,
-    ].flatten.join('_')
+      def destination
+        table, columns = *arg_groups.first
+        c = {
+          '+' => 'to',
+          '-' => 'from',
+          '%' => 'of'
```

Comment:
__[rubocop]__
* Style/TrailingCommaInLiteral: Put a comma after the last item of a multiline hash. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-array-commas)